src/wallet/wallet.cpp

```diff
@@ -2529,14 +2529,18 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             if (!out.fSpendable)
                  continue;
             nValueRet += out.tx->tx->vout[out.i].nValue;
-            setCoinsRet.insert(CInputCoin(out.tx->tx, out.i));
+            CInputCoin coin(out.tx->tx, out.i);
+            coin.effective_value = coin.txout.nValue - (out.nInputBytes < 0 || !use_effective ? 0 : coin_selection_params.effective_fee.GetFee(out.nInputBytes));
```

Comment:
nit: explicit precedence would be nice here: `(out.nInputBytes < 0 || !use_effective)`

Reply:
I thought I did this earlier. Must have gotten lost in a rebase. Fixed.

Reply:
Doesn't look changed fyi.